agent/engine/docker_task_engine.go

```diff
@@ -1556,7 +1556,7 @@ func (engine *DockerTaskEngine) pullAndUpdateContainerReference(task *apitask.Ta
 	if imageDigest != "" {
 		// If image digest is available then prepare a canonical reference to be used for image pull.
 		// This ensures that the image version referenced by the digest is pulled.
-		canonicalRef, err := referenceutil.GetCanonicalRef(imageRef, imageDigest)
+		canonicalRef, err := referenceutil.GetCanonicalRef(imageRef, imageDigest, false)
```

Comment:
why do we sometimes _not_ want to remove the tag?

Reply:
It's not necessary to remove the tag for imageref in this case, but we could since it won't affect anything.